.github/workflows/test-internal-prs.yml

```diff
@@ -0,0 +1,169 @@
+# Description: This workflow runs test suite against PRs that are not from forks.
```

Comment:
Well... this is supposed to a rename of `test-prs.yml` but Git is not recognizing it. 

The changes are:
1. Add codeql and dependency-review
2. Add status checks (to act as a PR merge requirement)